hooks/README.md

```diff
@@ -0,0 +1,98 @@
+# Camio Hooks
+
+## Introduction
+
+Camio provides an image processing pipeline. Videos are uploaded to the Camio cloud from
+a phone or an RTSP camera or a Camio Box. The Camio image pipeline provides a labeling 
+service but also allows you to register your own labeling hooks. A hook is defined as a custom web service that can compute labels from images.
+
+As videos are uploaded to Camio, they are broken into events,
+which are classified as boring, motion, and interesting. Interesting events are pre-classified by the Camio service, labeled, and a significative images are extracted from the videos in the event. If you have registsred a hook, and the event labels match your hook filter, the images beloging to the event are POSTed to your hook URL in the form of a JSON payload. Your hook URL should record the JSON payload, decoded it, extract the images, label them, and then POST the labels back to Camio at the callback_url provided by the initial hook call. The new labels are added to the event.
+
+Remember your hook labels events, not images, or videos. One event may contain mutiple images and multiple videos. The same video can span multiple consecutive events. At this time we only send to your hook preprocessed images. Also, you only label events that Camio has alreday pre-filtered for you as interesting.
+
+## Installing the example hook code
+
+The attached code include an installation script for Debian/Ubuntu 
+
+   hook-install.sh
+
+and an example of a hook:
+
+   hook-example.py
+
+You can run the install with 
+
+   ./hook-install.sh
+
+and it will install the required dependencies and start two processes:
+
+- a web server that received the hook calls and enqueues requests
+- a background process that handles the requests and posts back labels to Camio
+
+The queue is implemented as a mongodb database collection.
+
+## The example hook
+
+Our hook-example.py depends on bottle (0.13), gunicorn, requests, PIL, and pymongo.
+
+The hook-example.py code is build on bottle.py 0.13 and intended to work any WSGI web 
+server. We recommend gunicorn (a prefork WSGI server) and our install script assumes it.
+Once you have all the dependencies installed you can start the web server with:
+
+    nohup gunicorn -w 2 app:hook-example -b 0.0.0.0:8000 > /tmp/gunicorn.log &
+
+Here 8000 is the post to be used and 0.0.0.0 refers to (any IP address). 
+-w 2 requests two web server workers and /tmp/gunicorn.log is the location of the logfile.
+
+The server will expose three endpoints:
+
+- http://{your domain}:8000/tasks/ (GET) which you can call to check the service is running
+- http://{your domain}:8000/tasks/{api_key} (POST) which will you have to register with Camio and be called by Camio to post event images
```

Comment:
We need to make our variable placeholders consistent across all doc so that we choose either {param}, {{param}}, $param, or :param. Since we surface mustaches to end users, my slight preference is to standardize on {{}} but don't care as long as we're consistent.